MoreLinq/OrderedMerge.cs

```diff
@@ -98,7 +98,7 @@ static partial class MoreEnumerable
             IEnumerable<T> second,
             Func<T, TKey> keySelector)
         {
-            return OrderedMerge(first, second, keySelector, a => a, b => b, (a, _) => a, null);
+            return OrderedMerge(first, second, keySelector, Identity<T>, Identity<T>, (a, _) => a, null);
```

Comment:
Type argument is redundant:

```suggestion
            return OrderedMerge(first, second, keySelector, Identity, Identity, (a, _) => a, null);
```